### ISSUE TITLE
Allow specifying package manager during stylelint initialization

### ISSUE THREAD

**Issue by idkgene** [stylelint/stylelint#8227]
### What is the problem you're trying to solve?

Currently, when initializing a new stylelint configuration using `npm init stylelint`, the process implicitly assumes the use of npm as the package manager. This can be inconvenient for projects that utilize other package managers like pnpm or yarn. For instance, as demonstrated in the provided terminal output, when initializing ESLint with @eslint/config, the tool explicitly asks "Which package manager do you want to use?".

When stylelint initializes and installs dependencies, it uses npm. If a project is using pnpm, for example, the developer then needs to manually manage the dependencies. This extra step can be streamlined.

**1. Example eslint init output:**

```
npm init @eslint/config@latest
Need to install the following packages:
@eslint/create-config@1.4.0
Ok to proceed? (y) y


> destination@0.0.0 npx
> create-config

@eslint/create-config: v1.4.0

√ How would you like to use ESLint? · problems    
√ What type of modules does your project use? · esm
√ Which framework does your project use? · none
√ Does your project use TypeScript? · typescript
√ Where does your code run? · browser
The config that you've selected requires the following dependencies:

eslint, globals, @eslint/js, typescript-eslint
√ Would you like to install them now? · No / Yes
√ Which package manager do you want to use? · pnpm
☕️Installing...
```

**2. Example stylelint init output:**

```
> destination@1.0.0 npx
> create-stylelint

✔ Created .stylelintrc.json.
✔ Installed packages.

You can now lint your CSS files using:
npx stylelint "**/*.css"

We recommend customizing Stylelint:
https://stylelint.io/user-guide/customize/
```

### What solution would you like to see?

I propose adding an interactive option during the npm init stylelint process that allows users to select their preferred package manager. This could be implemented as a prompt similar to the one used by @eslint/create-config.

```
Which package manager do you want to use for installing dependencies?
  npm
> pnpm
  yarn
```

**Comment by idkgene** [comment]
The [@eslint/create-config utility](https://github.com/eslint/create-config/blob/a1d14172327f1bc9233a7896dd9cf5ddf1091a60/lib/utils/npm-utils.js#L53C1-L64C2) provides a good reference implementation for this. Specifically, it uses the installSyncSaveDev function to handle package installations, allowing the user's choice of package manager to be respected. Adopting a similar approach in stylelint would greatly improve the UXfor those not using npm.

I would be happy to contribute a Pull Request to implement this feature.

**Comment by Mouvedia** [comment]
@jeddy3 can you move this issue to stylelint/create-stylelint?

**Comment by idkgene** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/jeddy3?size=32">@jeddy3](https://github.com/jeddy3) can you transfer this issue to stylelint/create-stylelint?

Yep, I've found I've posted this in the wrong repo. If you'd like I can post it in there with PR referencing it and we just close this.
